numpy/lib/tests/test_format.py

```diff
@@ -628,6 +629,36 @@ def test_pickle_disallow():
     assert_raises(ValueError, np.save, path, np.array([None], dtype=object),
                   allow_pickle=False)
 
+@pytest.mark.parametrize('dt', [
+    np.dtype(np.dtype([('a', np.int8),
+                       ('b', np.int16),
+                       ('c', np.int32),
+                      ], align=True),
+             (3,)),
+    np.dtype([('x', ([('a', '|i1'),
+                      ('', '|V3'),
+                      ('b', '|i1'),
+                      ('', '|V3'),
+                     ],
+                     (3,)),
+               (4,),
+             )]),
+    np.dtype([('x', np.dtype({'names':['a','b'],
+                              'formats':['i1','i1'],
+                              'offsets':[0,4],
+                              'itemsize':8,
+                             },
+                    (3,)),
```

Comment:
I don't think `dtype(dict, tuple)` is legal, which will cause an error during test collection

Reply:
tests are passing

Reply:
```
>>> np.dtype(int, "this argument is ignored")
dtype('int32')
```

This test is ignoring the `(3,)` silently, which is a different bug.